File: server/src/main/java/de/ketrwu/portfolio/config/ApplicationConfiguration.java
package de.ketrwu.portfolio.config;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;

@Data
@Configuration
@ConfigurationProperties(prefix = "app")
public class ApplicationConfiguration {

    private Address address;

    private String[] domains;

    private String googleSiteVerification;

    private CloudProxyConfiguration cloud;

    @Data
    public static class Address {

        private String name;
        private String street;
        private String zip;
        private String country;

    }

    @Data
    public static class CloudProxyConfiguration {

        private String url;
        private String token;

    }
}

File: server/src/main/kotlin/de/ketrwu/portfolio/config/ApplicationConfiguration.kt
package de.ketrwu.portfolio.config

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.context.annotation.Configuration

/**
 * Configuration of application internal settings
 * @author Kenneth Wußmann
 */
@Configuration
@ConfigurationProperties(prefix = "app")
open class ApplicationConfiguration {
    var address: Address? = null
    var domains: Array<String>? = emptyArray()
    var googleSiteVerification: String? = null

    /**
     * Configuration for the address shown in imprint and privacy
     * @author Kenneth Wußmann
     */
    class Address {
        var name: String? = null
        var street: String? = null
        var zip: String? = null
        var country: String? = null
    }
}